### ISSUE TITLE
enforce inter stage dependency

### ISSUE THREAD

**Issue by AliTajeldin** [TresAmigosSD/SMV#147]
Best practice dictates that we use an InputSet to define the input interface to a stage and mark the output modules with `SmvOutput`.  However, there is nothing preventing users from depending on a module in another stage or using an SmvFile that is defined in another stage inputSet.
The inter stage module dependency is easy and logical to enforce once #146 is complete.

TBD: need to determine if we should also enforce SmvFile inter stage dependency.  That is, can one stage use the SmvFile val defined in another stage?

If an SmvFile *MUST* be defined in the same module that it is used, then two things could happen.
1. users will copy-n-paste the SmvFile definition (this would be hard to detect but not impossible).
2. we need to come up with an App level list of input files and then each stage InputSet would then reference the val defined in the app level list.  This would mean each file would require two entries to be updated when added.  Not sure this is nice.  The one advantage of this is the fact that the stage InputSet defines *ALL* the modules/files required by this stage.  The stage can not access anything without going through InputSet.

## Implementation Notes
To figure out where an SmvFile is defined, we would:
* rename SmvFile to SmvFileImpl
* create an SmvFile macro that would emit an SmvFileImpl instance and inject it with the enclosing InputSet.
* Since the enclosing InputSet can determine which module it is in, we can determine the SmvFile module.

Note: Quite a bit of stuff that needs to determined and discussed before any of this is implemented.

**Comment by AliTajeldin** [comment]
Note that the concept of InputSet is going away, but we can still do the check on module/file inter stage dependency.

**Comment by AliTajeldin** [comment]
Closing as duplicate of #167 as that one does not have all the InputSet overhead.
